### PR TITLE
Add CO-RE bitfield support to `bindsnoop`

### PR DESCRIPTION
**Pull request by eiffel-fl**
Hi.

This PR uses a new feature from `cilium/ebpf` to be able to print port and options of a `bind()`.

Best regards.

### CODE REVIEW

**Comment by eiffel-fl** [comment]
I reworked this PR to use `cilium/ebpf` master on a specific commit since cilium/ebpf#517 was merged.
But I am not forcefully satisfied with how I handled it, so your reviews are welcomed.
Particularly, I do not know if we should wait for a cilium/ebpf release to use and I do not like how I had to deal with CI cache.

File: pkg/gadgettracermanager/gadgettracermanager.go
Comment (mauriciovasquezbernal) [inline pkg/gadgettracermanager/gadgettracermanager.go:213]: I think we should use [`RemoveMemlock()`](https://github.com/cilium/ebpf/blob/v0.8.1/rlimit/rlimit.go#L104), it has some additional logic to be no-op when no needed.

File: .github/actions/install-debian-packages/action.yml
Comment (mauriciovasquezbernal) [inline .github/actions/install-debian-packages/action.yml:24]: What about updating the `key` used in `actions/cache@v2`? 

Btw, now that I look with more detail at the logic to install the deb packages, isn't it possible to move all usages of `actions/cache@v2` from `workflows/inspektor-gadget.yml` to `install-debian-packages/action.yml` to avoid repeating that so many times?

**Review by mauriciovasquezbernal — COMMENTED** [review]
Lest some comments. Please organize the commits a bit, commit messages are not complete and the don't follow a logic sequence.

**Comment by eiffel-fl** [comment]
>Please organize the commits a bit, commit messages are not complete and the don't follow a logic sequence.

I am also not satified with them and I do not really know how to organize them...
The only good way to do it would be:
1. One commit to bump cilium/ebpf version which also adapt the code to upstream changes.
2. Another which does the revert.
Do you see a better option?

File: pkg/gadgettracermanager/gadgettracermanager.go
Comment (eiffel-fl) [reply pkg/gadgettracermanager/gadgettracermanager.go:213]: Good catch, I did not see this function existed hence going `unix`.

**Comment by mauriciovasquezbernal** [comment]
>     1. One commit to bump cilium/ebpf version which also adapt the code to upstream changes.
> 
>     2. Another which does the revert.
>        Do you see a better option?

Yes, that should work fine.

File: .github/actions/install-debian-packages/action.yml
Comment (mauriciovasquezbernal) [inline .github/actions/install-debian-packages/action.yml:20]: Perhaps we can add a comment here indicating this should be updated when the packages are changed. Just to avoid somebody else hitting the same problem you hit with the cache.

Comment (eiffel-fl) [reply .github/actions/install-debian-packages/action.yml:20]: I wrote it here:
https://github.com/kinvolk/inspektor-gadget/pull/529/files/ced6dc7b5df83b9b0c66a1053bb4e62278d2be34#diff-6aa1982dd6b6db00df35c70d94604a381bf22ea49c53a97aeb03bcd05e1a74dbR27
But I can put it over the key field if needed.

Comment (mauriciovasquezbernal) [reply .github/actions/install-debian-packages/action.yml:20]: I missed it. It seems a good place to have it there.

File: .github/actions/install-debian-packages/action.yml
Comment (mauriciovasquezbernal) [inline .github/actions/install-debian-packages/action.yml:21]: One additional question here. What is the procedure to update a dependency? Or better, how avoid this logic to always use the check instead of trying to install new versions of the libraries?

**Review by mauriciovasquezbernal — COMMENTED** [review]
I tested and it worked fine. Just one last question about debs cache mechanism in github actions.

File: .github/actions/install-debian-packages/action.yml
Comment (eiffel-fl) [reply .github/actions/install-debian-packages/action.yml:21]: As long as we do not change the cache key, we will be stuck with the version of the software which were installed for this given cache key.
This should not be a problem for `libbpf` and `libbseccomp` as we install specific version of this libraries, but for `llvm` it can be more troublesome...
We will need to think to a better way to handle this case while not losing the speed gain of the cache.

Comment (mauriciovasquezbernal) [reply .github/actions/install-debian-packages/action.yml:21]: You're right. Let's think about a better way to handle it in another PR.

**Review by mauriciovasquezbernal — APPROVED** [review]
LGTM! Please notice we should wait the release of v0.5.0 before merging it.

**Comment by eiffel-fl** [comment]
Thank you for the review!
I will hold it until.

**Comment by marga-kinvolk** [comment]
0.5.0 has been released, do you want to merge this now?

**Comment by eiffel-fl** [comment]
Le mercredi 27 avril 2022, 13:51:06 BST Marga Manterola a écrit :
> 0.5.0 has been released, do you want to merge this now?

Yes! This is why I rebased it but I forgot to check the results of the CI 
yesterday :sweat_smile:.

**Comment by eiffel-fl** [comment]
Thank you for the reviews!

**Pull request merged**
